src/modules/xworkspaces.cpp

```diff
@@ -20,6 +20,15 @@ namespace {
   }
 }  // namespace
 
+/**
+ * Defines a lexicographical order on position:
+ *
+ * (x1, y1) < (x2, y2) <==> (x1 == x2 && y1 < y2) || (x1 < x2)
+ */
+bool operator<(const position& a, const position& b) {
+  return a.x < b.x || (a.x == b.x && a.y < b.y);
```

Comment:
For this case this might be overkill but if you want to do that for a struct that have a lot of attributes.

```suggestion
  return std::make_tuple(a.x, a.y) < std::make_tuple(b.x, b.y);
```